src/components/approval-type/ApprovalType.vue

```diff
@@ -64,53 +105,86 @@ export default class ApprovalType extends Vue {
   /** Show only the court order option; remove via registrar option. */
   @Prop({ default: false }) readonly isCourtOrderOnly!: boolean
 
+  /** Draft notice date. */
+  @Prop({ default: '' }) readonly notice!: string
+
+  /** Draft application date. */
+  @Prop({ default: '' }) readonly application!: string
+
   // Local properties
   private courtOrderNumber = ''
   private courtOrderNumRules = []
   private valid = false
   private approvalTypeSelected = ''
+  private datePickerRules = []
+  private noticeDateText = ''
+  private applicationDateText = ''
 
   /** Called when component is mounted. */
   mounted (): void {
+    this.setDateRules()
+    this.setCourtNumberRules()
     // Copy props to mutable properties
     if (this.draftApprovedByRegistrar) {
       this.courtOrderNumber = ''
       this.approvalTypeSelected = this.ApprovalTypes.VIA_REGISTRAR
-    }
-    if (this.draftCourtOrderNumber) {
+    } else if (this.draftCourtOrderNumber) {
       this.courtOrderNumber = this.draftCourtOrderNumber
       this.approvalTypeSelected = this.ApprovalTypes.VIA_COURT_ORDER
+    } else {
+      // Default state (no button selected)
+      this.radioButtonChanged('')
     }
   }
 
   /** Triggers the form validation. */
   public validate (): boolean {
-    let status = this.$refs.courtNumRef.validate()
-    this.$emit('emitValid', status)
-    return status
+    if (this.approvalTypeSelected === this.ApprovalTypes.VIA_COURT_ORDER) {
+      let status = this.$refs.courtNumRef.validate()
+      this.$emit('emitValid', status)
+      return status
+    } else if (this.approvalTypeSelected === this.ApprovalTypes.VIA_REGISTRAR) {
+      // Emit true (valid) if both dates were selected. Emit false (invalid) if at least one was empty.
+      let status = (!!this.noticeDateText && !!this.applicationDateText)
+      this.$emit('emitValid', status)
+      return status
+    } else {
+      // Default state. Emit false (invalid).
+      return false
+    }
+  }
+
+  /** Validations rules for the date picker fields. */
+  private setDateRules (): void {
+    this.datePickerRules = [
+      (v: string) => !!v || 'Date is required'
+    ]
   }
 
   private emitValidationError (event: boolean): void {
     this.$emit('emitValid', !event)
   }
 
+  // Emit the approval type (radio button selected).
   @Emit('emitRadioButtonChange')
   private radioButtonChanged (event: string): void {
     if (event === this.ApprovalTypes.VIA_REGISTRAR) {
       Vue.set(this, 'courtOrderNumber', '')
       Vue.set(this, 'courtOrderNumRules', [])
-    }
-    if (event === this.ApprovalTypes.VIA_COURT_ORDER) {
+    } else if (event === this.ApprovalTypes.VIA_COURT_ORDER) {
       this.setCourtNumberRules()
```

Comment:
First, this was already called in mounted().

Second, it would not need to get called just in this case (VIA_COURT_ORDER) since there is no parallel call to unset the rules. In order words, they're just always declared.

So, I recommend just declaring them statically (like the date rules per my other comment).